src/file-table-engine/src/engine/tests.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::assert_matches::assert_matches;
+
+use common_catalog::consts::{DEFAULT_CATALOG_NAME, DEFAULT_SCHEMA_NAME, IMMUTABLE_FILE_ENGINE};
+use table::engine::{EngineContext, TableEngine, TableReference};
+use table::requests::{AlterKind, AlterTableRequest, DropTableRequest, OpenTableRequest};
+use table::{error as table_error, Table};
+
+use crate::config::EngineConfig;
+use crate::engine::immutable::ImmutableFileTableEngine;
+use crate::manifest::immutable::manifest_path;
+use crate::table::immutable::ImmutableFileTable;
+use crate::test_util::{self, TestEngineComponents, TEST_TABLE_NAME};
+
+#[tokio::test]
+async fn test_get_table() {
+    let TestEngineComponents {
+        table_engine,
+        table_ref: table,
+        dir: _dir,
+        ..
+    } = test_util::setup_test_engine_and_table("test_get_table").await;
+    let table_info = table.table_info();
+    let table_ref = TableReference {
+        catalog: &table_info.catalog_name,
+        schema: &table_info.schema_name,
+        table: &table_info.name,
+    };
+
+    let got = table_engine
+        .get_table(&EngineContext::default(), &table_ref)
+        .unwrap()
+        .unwrap();
+
+    assert_eq!(table.schema(), got.schema());
+}
+
+#[tokio::test]
+async fn test_open_table() {
```

Comment:
I think we should rewrite the open test to the following form:
- close the table
- open the table again
- checks the metadata

It's more clear. But we need to add a `close_table()` method to the ImmutableFileTableEngine in order to close and then remove the table from the engine inner. (Actually, we should provide this method in the table engine trait, but it's unrelated to this PR).